### PR TITLE
Fix updates of stateful set images

### PR DESCRIPTION
**Pull request by pdecat**
Images used for container in Stateful Set templates cannot be updated.

TODO:
- [x] add acceptance test
- [x] fix issue

Acceptance test result without fix:
```
# make testacc TEST=./kubernetes TESTARGS='-run=TestAccKubernetesStatefulSet_update_image -count=1'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./kubernetes -v -run=TestAccKubernetesStatefulSet_update_image -count=1 -timeout 120m
=== RUN   TestAccKubernetesStatefulSet_update_image
--- FAIL: TestAccKubernetesStatefulSet_update_image (3.01s)
        testing.go:518: Step 1 error: Check failed: 1 error(s) occurred:

                * Check 2/2 error: kubernetes_stateful_set.test: Attribute 'spec.0.template.0.spec.0.container.0.image' expected "k8s.gcr.io/liveness:latest", got "k8s.gcr.io/pause:latest"
FAIL
FAIL    github.com/terraform-providers/terraform-provider-kubernetes/kubernetes 3.059s
make: *** [GNUmakefile:17: testacc] Error 1
```

### CODE REVIEW

**Comment by pdecat** [comment]
Acceptance test results with fix:

```
# make testacc TEST=./kubernetes TESTARGS='-run=TestAccKubernetes\(StatefulSet\).* -count=1'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./kubernetes -v -run=TestAccKubernetes\(StatefulSet\).* -count=1 -timeout 120m
=== RUN   TestAccKubernetesStatefulSet_basic
--- PASS: TestAccKubernetesStatefulSet_basic (1.81s)
=== RUN   TestAccKubernetesStatefulSet_basic_idempotency
--- PASS: TestAccKubernetesStatefulSet_basic_idempotency (1.94s)
=== RUN   TestAccKubernetesStatefulSet_update_image
--- PASS: TestAccKubernetesStatefulSet_update_image (2.88s)
=== RUN   TestAccKubernetesStatefulSet_update_template_selector_labels
--- PASS: TestAccKubernetesStatefulSet_update_template_selector_labels (2.64s)
=== RUN   TestAccKubernetesStatefulSet_update_replicas
--- PASS: TestAccKubernetesStatefulSet_update_replicas (2.40s)
=== RUN   TestAccKubernetesStatefulSet_update_rolling_update_partition
--- PASS: TestAccKubernetesStatefulSet_update_rolling_update_partition (2.40s)
=== RUN   TestAccKubernetesStatefulSet_update_update_strategy_on_delete
--- PASS: TestAccKubernetesStatefulSet_update_update_strategy_on_delete (2.35s)
=== RUN   TestAccKubernetesStatefulSet_update_update_strategy_rolling_update
--- PASS: TestAccKubernetesStatefulSet_update_update_strategy_rolling_update (2.35s)
=== RUN   TestAccKubernetesStatefulSet_update_pod_template_container_port
--- PASS: TestAccKubernetesStatefulSet_update_pod_template_container_port (2.59s)
PASS
ok      github.com/terraform-providers/terraform-provider-kubernetes/kubernetes 21.400s
```

**Comment by pdecat** [comment]
The parameters passed to `patchPodSpec` were swapped: https://github.com/terraform-providers/terraform-provider-kubernetes/pull/252/files#diff-14a25dc6896cc8eaf7d1a5bcaa4da883R270

@alexsomesan PTAL

File: test.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+apiVersion: v1
```
Comment (alexsomesan) [inline test.yml]: Is this file unintentionally left over from local manual testing?

Comment (pdecat) [reply test.yml]: Oops :) Fixed!

**Pull request merged**